pwiz_tools/Skyline/Model/Export.cs

```diff
@@ -2986,197 +3017,284 @@ private static string GetModuleName(Process proc)
         }
     }
 
-    public class AgilentMassListExporter : AbstractMassListExporter
+    public abstract class AgilentMassListExporter : AbstractMassListExporter
     {
-        public AgilentMassListExporter(SrmDocument document)
-            : this(document, null)
+
+        public class TransitionContext
         {
+            public AgilentMassListExporter parent;
+
+            public PeptideGroupDocNode nodePepGroup;
+            public PeptideDocNode nodePep;
+            public TransitionGroupDocNode nodeTranGroup;
+            public TransitionGroupDocNode nodeTranGroupPrimary;
+            public TransitionDocNode nodeTran;
+            public int step;
         }
 
-        public AgilentMassListExporter(SrmDocument document, DocNode node)
-            : base(document, node)
+        public class TransitionField

```

Comment:
Instead of defining a list of functions that take a TransitionContext and return a value, and then constructing TransitionField's from those function, I would recommend making it so that TransitionContext has public Properties on it for each of these fields, and then use reflection to list those properties.

Let me know if you'd like me to write some example code for this.